### PR TITLE
Fix metadata TFTP

### PR DESCRIPTION
**Pull request by LipuFei**
Fix metadata download through TFTP.

Thumbnails are stored in LevelDb. The key is the SHA1 hash of the thumbnail file.

### CODE REVIEW

**Comment by LipuFei** [comment]
retest this please

**Comment by LipuFei** [comment]
@whirm ready

**Comment by LipuFei** [comment]
@whirm wait. perhaps after the Plyvel PR.

File: Tribler/Core/SessionConfig.py [hunk 1]
```diff
@@ -608,6 +608,48 @@ def set_enable_channel_search(self, mode):
         """
         self.sessconfig.set(u'allchannel_community', u'enabled', mode)
 
+    def get_enable_metadata_community(self):
```
Comment (NielsZeilemaker) [inline Tribler/Core/SessionConfig.py]: Is the metadata community making a comeback?

Comment (LipuFei) [inline Tribler/Core/SessionConfig.py]: @NielsZeilemaker I will probably just use the Channel community for metadata propagation. This PR is to fix the metadata download via TFTP only so I haven't modified the metadata community part yet.

Comment (NielsZeilemaker) [inline Tribler/Core/SessionConfig.py]: But, you're adding these new methods here right? Why did you add them?

Comment (LipuFei) [inline Tribler/Core/SessionConfig.py]: Good point... I will change it to something like "enable_metadata" instead.

**Comment by LipuFei** [comment]
@NielsZeilemaker The `enable_metadata_community` option is no more.

**Comment by NielsZeilemaker** [comment]
Seems ready to me

**Comment by LipuFei** [comment]
@NielsZeilemaker @whirm Let's wait for #1488. I will do a rebase after it.

File: Tribler/Core/leveldbstore.py [hunk 2]
```diff
@@ -50,11 +50,11 @@
 # it if it doesn't as it complicates the code.
```
Comment (whirm) [inline Tribler/Core/leveldbstore.py]: can you remove this todo entry :D ?  
```
# TODO(emilon): This could be easily abstracted into a generic cached store
```

Comment (LipuFei) [inline Tribler/Core/leveldbstore.py]: @whirm Done.

**Comment by whirm** [comment]
that last failure is actually my fault, I introduced a bug in the non-parallel test execution. Fixing it.

**Comment by whirm** [comment]
should be fine now

**Comment by LipuFei** [comment]
@whirm ready when it passes

**Comment by whirm** [comment]
@LipuFei looks fine to me

**Comment by LipuFei** [comment]
@whirm Should be ready then :)
